src/backend/cdb/cdbquerycontextdispatching.c

```diff
@@ -616,6 +632,37 @@ RebuildFilesystemCredentials(QueryContextInfo *cxt, HTAB ** currentFilesystemCre
 	pfree(binary);
 }
 
+
+/*
+ * deserialize the namespace data
+ */
+
+static void
+RebuildNamespace(QueryContextInfo *cxt)
+{
+
+	int len;
+	char buffer[4], *binary;
+	ReadData(cxt, buffer, sizeof(buffer), TRUE);
+
+	len = (int) ntohl(*(uint32 *) buffer);
+	binary = palloc(len);
+	if(ReadData(cxt, binary, len, TRUE))
+	{
+		StringInfoData buffer;
+		initStringInfoOfString(&buffer, binary, len);
+
+		cxt->sharedPath = palloc(buffer.len + 1);
+		while(buffer.cursor < buffer.len) {
```

Comment:
missing space after while.
Also I think the palloc + loop can be replaced by a copy function `pnstrdup`